### DISCUSSION TITLE
Looser coupling for projects in the SDK

### DISCUSSION THREAD

**Discussion by dbsanfte** [Infisical/infisical#6311] [Feature Request]
### Feature description
Presently in e.g. the Python SDK you need to couple tightly to a project by its GUID:

![image](https://github.com/user-attachments/assets/d026d11e-dc15-4498-8f64-38e24d6e040d)

It would be nice if you could couple loosely by its name instead.

`project_name="my-project-name"`

### Why would it be useful?
Less intervention by users/myself to fetch a GUID when a project name is all I  should need. I'm thinking of Golden Template clones in a Github repo, where I know the project name because it should match the Github repo name... but no, I need to go to Infisical to fetch an arbitrary GUID ID. 😢 

### Additional context
My developers raised this to me.

**Comment by akhilmhdh** [comment]
Moving this to [feature request discussion](https://github.com/Infisical/infisical/issues/5172).
